packages/rulesets/src/spectral/functions/provisioning-state-must-be-read-only.ts

```diff
@@ -0,0 +1,26 @@
+import { getProperty, getProperties } from "./utils"
+
+export const provisioningStateMustBeReadOnly = (schema: any, _opts: any, ctx: any) => {
+  if (schema === null || typeof schema !== "object") {
+    return []
+  }
+
+  const path = ctx.path || []
+  const errors = []
+
+  const allProperties = getProperties(schema)
+  const provisioningStateProperty = getProperty(allProperties?.properties, "provisioningState")
```

Comment:
Looking at the utils functions, it looks like you can just use `getProperty` directly, and do not need to use `getAllProperties` in this case.
```suggestion
  const provisioningStateProperty = getProperty(schema, "provisioningState")
```

Reply:
I used the same in one of my previous rules & it started false negatives as ProvisioningState property is within the properties bucket in properties, so we do one level deep and then get the property